contracts/Controller.sol

```diff
@@ -158,38 +158,43 @@ contract Controller is Ownable {
         address oracleModule = AddressBookInterface(addressBook).getOracle();
         OracleInterface oracle = OracleInterface(oracleModule);
 
-        OtokenInterface otoken = OtokenInterface(_otoken);
-
-        address otokenUnderlyingAsset = otoken.underlyingAsset();
-        address otokenStrikeAsset = otoken.strikeAsset();
-        address otokenCollateralAsset = otoken.collateralAsset();
-        uint256 otokenExpiryTimestamp = otoken.expiryTimestamp();
-
-        bytes32 batch = keccak256(
-            abi.encode(otokenUnderlyingAsset, otokenStrikeAsset, otokenCollateralAsset, otokenExpiryTimestamp)
-        );
-
+        (bytes32 batch, , , , uint256 otokenExpiryTimestamp) = getBatchDetails(_otoken);
         return oracle.isDisputePeriodOver(batch, otokenExpiryTimestamp);
     }
 
     /**
      * @notice get batch for a specific otoken address
      * @dev batch is the hash of option underlying, strike, collateral assets and the expiry timestamp
      * @param _otoken otoken address
-     * @return batch hash in bytes32
+     * @return batch hash in bytes32, batch underlying asset, batch strike asset, batch collateral asset, batch expiry timestamp
      */
-    function getBatch(address _otoken) external view returns (bytes32) {
+    function getBatchDetails(address _otoken)
+        public
+        view
+        returns (
+            bytes32,
+            address,
+            address,
+            address,
+            uint256
```

Comment:
can you name these variables, otherwise it's hard to get any information from this function signature

Reply:
We usually don't name return veriables, let't keep it that way as the name are in the return statement IMO.